### PR TITLE
tweaks to New-TestResources readme

### PR DESCRIPTION
**Pull request by christothes**

### CODE REVIEW

File: eng/common/TestResources/README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ Connect-AzAccount -Subscription 'YOUR SUBSCRIPTION ID'
 eng\common\TestResources\New-TestResources.ps1 -ServiceDirectory 'search'
 ```
 
-The `OutFile` switch would be set if you are running this for a .NET project on Windows. This will save test environment settings
+The `OutFile` switch should be set if you are running this for a .NET project on Windows. This will save test environment settings
```
Comment (heaths) [inline eng/common/TestResources/README.md:33]: This should stay "would" or even more firmly be "will". It defaults to `$true` on Windows. "should" makes it sound like "you should set this".

File: eng/common/TestResources/README.md [hunk 2]
```diff
@@ -89,6 +89,12 @@ Remove-AzADServicePrincipal -ApplicationId $sp.ApplicationId -Force
 
 If you persisted environment variables, you should also remove those as well.
 
+Some scripts accept the `AdditionalParameters` parameter to control additional resource configuration options. For example:
+
+```powershell
+.\eng\common\TestResources\New-TestResources.ps1 keyvault -AdditionalParameters @{enableHsm = $true}
```
Comment (heaths) [inline eng/common/TestResources/README.md:95]: Like other samples, don't include the path. This assumes too much of the CWD of users. It's not common in PowerShell documentation in general.

**Review by heaths — APPROVED** [review]

File: eng/common/TestResources/README.md [hunk 3]
```diff
@@ -89,6 +89,12 @@ Remove-AzADServicePrincipal -ApplicationId $sp.ApplicationId -Force
 
 If you persisted environment variables, you should also remove those as well.
 
+Some scripts accept the `AdditionalParameters` parameter to control additional resource configuration options. For example:
```
Comment (benbp) [inline eng/common/TestResources/README.md:92]: Glad this is being specified, though I'm not sure of the wording. What does "some scripts" refer to in this case?

I think you could be more specific that these parameters end up becoming parameters to the ARM template.

Comment (christothes) [reply eng/common/TestResources/README.md:92]: Good call out.

**Review by benbp — APPROVED** [review]

**Comment by christothes** [comment]
@benbp  Should I override check enforcer on this?

**Comment by benbp** [comment]
@christothes you need to click through [the pipeline](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=710526&view=results) and manually approve a few things. It needs to: create sync PRs for your change against all the repos, then you have to wait until all those are green (and approve them yourself), then approve the pipeline against which will merge them.

You can see the workflow docs here: https://github.com/Azure/azure-sdk-tools/blob/master/doc/common/common_engsys.md#workflow

**Comment by christothes** [comment]
@benbp The js pipeline is failing for apparently unrelated issues. I attempted to re-run a few times just to check if it was transient.

**Comment by weshaggard** [comment]
@christothes for the JS case if it the failure is unrelated just override check-enforcer and merge manually.

BTW it looks like you manually merged l the sync PR's. For future reference you only need to approve each of those PR's and then the automation in the "azure-sdk-tools - sync - eng-common" pipeline will attempt to merge all the PRs.

**Pull request merged**